tests/e2e/test_run_demos.py

```diff
@@ -0,0 +1,192 @@
+import logging
+import time
+
+from pathlib import Path
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.workspace import ImportFormat
+from databricks.sdk.service.pipelines import NotebookLibrary, PipelineLibrary, UpdateInfoState
+from databricks.sdk.service.jobs import NotebookTask, Task, RunLifecycleStateV2State, TerminationTypeType
+
+logging.getLogger("tests").setLevel("DEBUG")
+logging.getLogger("databricks.labs.dqx").setLevel("DEBUG")
+logger = logging.getLogger(__name__)
+
+
+RETRY_INTERVAL_SECONDS = 30
+
+
+def test_run_dqx_demo_library(make_notebook, make_schema, make_job):
+    path = Path(__file__).parent.parent.parent / "demos" / "dqx_demo_library.py"
+    ws = WorkspaceClient()
+    with open(path, "rb") as f:
+        notebook = make_notebook(content=f, format=ImportFormat.SOURCE)
+        directory = notebook.as_fuse().parent.as_posix()
+
+    catalog = "main"
+    schema = make_schema(catalog_name=catalog).name
+    notebook_path = notebook.as_fuse().as_posix()
+    notebook_task = NotebookTask(
+        notebook_path=notebook_path,
+        base_parameters={"demo_database_name": catalog, "demo_schema_name": schema, "demo_file_directory": directory},
+    )
+    job = make_job(tasks=[Task(task_key="dqx_demo_library", notebook_task=notebook_task)])
+    run = ws.jobs.run_now(job.job_id)
+
+    while True:
```

Comment:
Can we use waiter here? 

`result = run.result(timeout=datetime.timedelta(seconds=RETRY_INTERVAL_SECONDS))`

See https://github.com/databricks/databricks-sdk-py?tab=readme-ov-file#long-running-operations